src/payment/strategies/adyenv2/adyenv2-payment-strategy.ts

```diff
@@ -0,0 +1,289 @@
+
+        return storeConfig.storeProfile.storeLanguage;
+    }
+
+    private _getStateContainer(): string {
+        if (!this._stateContainer) {
+            return '{}';
+        }
+
+        return this._stateContainer;
+    }
+
+    private _getThreeDS2ChallengeWidgetSize(): string {
+        const { threeDS2ChallengeWidgetSize } = this._getAdyenV2PaymentInitializeOptions();
+
+        if (!threeDS2ChallengeWidgetSize) {
+            return '01';
+        }
+
+        return threeDS2ChallengeWidgetSize;
+    }
+
+    private _handle3DS2Challenge(resultObject: ThreeDS2Result, paymentMethodId: string): Promise<Payment> {
+
+        return new Promise((resolve, reject) => {
+            if (!this._adyenCheckout) {
+                throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+            }
+
+            const threeDS2Component = this._adyenCheckout
+                .create(ThreeDS2ComponentType.ThreeDS2Challenge, {
+                    challengeToken: resultObject.token,
+                    onComplete: (challengeData: ThreeDS2OnComplete) => {
+                        const challengePaymentPayload = {
+                            ...challengeData.data,
+                            paymentData: resultObject.payment_data,
+                        };
+
+                        const paymentPayload = {
+                            methodId: paymentMethodId,
+                            paymentData: {
+                                nonce: JSON.stringify(challengePaymentPayload, null, 2),
+                            },
+                        };
+
+                        resolve(paymentPayload);
+                    },
+                    onError: (error: AdyenError) => reject(error),
+                    size: this._getThreeDS2ChallengeWidgetSize(),
+                });
+
+            threeDS2Component.mount(`#${this._getAdyenV2PaymentInitializeOptions().containerId}-3ds`);
+        });
+    }
+
+    private _handle3DS2Fingerprint(resultObject: ThreeDS2Result, paymentMethodId: string): Promise<Payment> {
+
+        return new Promise((resolve, reject) => {
+            if (!this._adyenCheckout) {
+                throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+            }
+
+            const threeDS2Component = this._adyenCheckout
+                .create(ThreeDS2ComponentType.ThreeDS2DeviceFingerprint, {
+                    fingerprintToken: resultObject.token,
+                    onComplete: (fingerprintData: ThreeDS2OnComplete) => {
+                        const fingerprintPaymentPayload = {
+                            ...fingerprintData.data,
+                            paymentData: resultObject.payment_data,
+                        };
+
+                        const paymentPayload = {
+                            methodId: paymentMethodId,
+                            paymentData: {
+                                nonce: JSON.stringify(fingerprintPaymentPayload, null, 2),
```

Comment:
If you don't need to add additional whitespace to the JSON output, I prefer you leaving out the 2nd and 3rd argument.